### PR TITLE
Don't use NCHW or NHCW in tf.layers.conv1d

### PR DESCRIPTION
**Pull request by alanhdu**
Fixes deprecation warning when using tf.layers.conv1d.

### CODE REVIEW

**Comment by alanhdu** [comment]
This was initially closed because @protoget thought that:

> Internally we still need NCHW or NWHC, the data_format is later passed to gen_nn_ops.conv2d.

But I've confirmed that this is actually not true: `tf.nn.conv1d` explicitly converts `NCW` and `NWC` back to `NCHW` and `NWHC`:

https://github.com/tensorflow/tensorflow/blob/6fdb9ad1baf7686a75f9e660178f7ac595e7fc2e/tensorflow/python/ops/nn_ops.py#L2428-L2435

and has explicitly deprecated `NCHW` and NWHC` as arguments:

https://github.com/tensorflow/tensorflow/blob/b07680459a88224fce83daa7b3b70bcc62b9c896/tensorflow/python/ops/nn_ops.py#L2366-L2375

File: tensorflow/python/ops/nn_ops.py [hunk 1]
```diff
@@ -149,14 +149,12 @@ def __init__(
...
     if conv_dims == 1:
       # conv1d uses the 2-d data format names
-      if data_format is None or data_format == "NWC":
-        data_format_2d = "NHWC"
-      elif data_format == "NCW":
-        data_format_2d = "NCHW"
-      else:
+      if data_format is None:
+        data_format = "NWC"
+      elif data_format not in {"NCW", "NWC"}:
```
Comment (reedwm) [inline tensorflow/python/ops/nn_ops.py]: Before it only gave a deprecation warning if the user passed "NCHW" or "NHWC" but now it raises an error. Change {"NCW", "NWC"} to {"NCW", "NWC", "NCHW", "CHWC"}

**Review by reedwm — CHANGES_REQUESTED** [review]

**Comment by alanhdu** [comment]
@reedwm Done!

**Review by reedwm — APPROVED** [review]

**Comment by alanhdu** [comment]
@reedwm @protoget Is there anything else I need to do here? I'm hoping that this can get merged in for the 1.7 release (if only for the aesthetic value of not having deprecation warnings in my code :stuck_out_tongue_winking_eye:).

**Comment by reedwm** [comment]
Sorry for the delay.

@gunan, will this be included in 1.7?

**Comment by gunan** [comment]
No, 1.7 is almost final.
And since this is only warnings, we will probably not delay the final release for this cherrypick.

**Pull request merged**